File: algoritmos/restar_sin_restar/ejemploV2.js
const subtractionV2 = (num1, num2) => {
  let result = 0;

  if (num1 > num2) {
    result = _incrementValue(num1, num2, 0);
  } else if (num2 > num1) {
    result = _incrementValue(num2, num1, 0);
    return result * -1;
  }
  return result;
};

const _incrementValue = (bigger, minor, value) => {
  while (bigger > minor) {
    minor++;
    value++;
  }
  return value;
};

File: algoritmos/restar_sin_restar/ejemploV2.py
def subtraction_v2(num1, num2):
    result = 0

    if num1 > num2:
        result = increment_value(num1, num2, 0)
    elif num2 > num1:
        result = increment_value(num2, num1, 0)
        return result * -1
    return result


def increment_value(num1, num2, value):
    while num1 > num2:
        num2 += 1
        value += 1
    return value


def run():
    pass


if __name__ == "__main__":
    run()
